openedx/core/djangoapps/user_api/tests/test_views.py

```diff
@@ -800,6 +803,99 @@ def setUp(self):
         super(RegistrationViewValidationErrorTest, self).setUp()
         self.url = reverse("user_api_registration")
 
+    def retireRequestUser(self):
+        """
+        Very basic user retirement initiation, logic copied form DeactivateLogoutView.  This only lands the user in
+        PENDING, simulating a retirement request only.
+        """
+        user = User.objects.get(username=self.USERNAME)
+        UserRetirementStatus.create_retirement(user)
+        user.email = get_retired_email_by_email(user.email)
+        user.set_unusable_password()
+        user.save()
+
+    @mock.patch('openedx.core.djangoapps.user_api.views.check_account_exists')
+    def test_register_retired_email_validation_error(self, dummy_check_account_exists):
```

Comment:
The check for retired emails added in common/djangoapps/student/forms.py AccountCreationForm.clean_email, which is...

* used by common/djangoapps/student/views/management.py in create_account_with_params()
* used by openedx/core/djangoapps/user_api/views.py in RegistrationView.post()
* tested in openedx/core/djangoapps/user_api/tests/test_views.py

Is duplicated by the retired emails check also added in check_account_exists.  (yes, we check twice for duplicate emails already, in this same transaction).  Since I updated both mechanisms for checking for duplicate emails, I decided to test both, hence the two tests.

Reply:
I see - so the mocking is done to allow the second check to be made and tested.